core/src/com/unciv/ui/civilopedia/CivilopediaText.kt

```diff
@@ -8,28 +8,66 @@ import com.badlogic.gdx.utils.Align
+ *      into the parent's children array. So, we're SOL to get the number of texture switches down
+ *      with this structure, as many lines will require at least 2 texture switches.
+ *      We *could* instead try go for one big table with 4 columns (3 images, plus rest)
+ *      and use colspan - then group all images separate from labels via ZIndex. To-Do later.
+ *    - Do bold using Distance field fonts wrapped in something like [maltaisn/msdf-gdx](https://github.com/maltaisn/msdf-gdx)
+ *    - Do strikethrough by stacking a line on top (as rectangle with background like the separator but thinner)
+ */
 
 
-/** Represents a text line with optional linking capability.
+/** Represents a decorated text line with optional linking capability.
+ *  A line can have up to three icons: link, object, star in that order.
  *  Special cases:
+ *  - Standalone image from atlas or from ExtraImages
+ *  - A separator line ([separator])
  *  - Automatic external links (no [text] but [link] begins with a URL protocol)
  * 
  * @param text          Text to display.
  * @param link          Create link: Line gets a 'Link' icon and is linked to either 
  *                      an Unciv object (format `category/entryname`) or an external URL.
+ * @param icon          Display an Unciv object's icon inline but do not link (format `category/entryname`).
  * @param extraImage    Display an Image instead of text. Can be a path as understood by
  *                      [ImageGetter.getImage] or the name of a png or jpg in ExtraImages.
  * @param imageSize     Width of the [extraImage], height is calculated preserving aspect ratio. Defaults to available width.
+ * @param size          Text size, defaults to 18f. Use [size] or [header] but not both.
  * @param header        Header level. 1 means double text size and decreases from there.
- * @param separator     Renders a separator line instead of text.
+ * @param indent        Indentation - 0 = text will follow icons with a little padding,
+ *                      1 = aligned to a little more than 3 icons, each step above that adds 30f.
+ * @param padding       Defines vertical padding between rows, defaults to 5f.
+ * @param color         Sets text color, accepts Java names or 6/3-digit web colors (e.g. #FFA040).
+ * @param separator     Renders a separator line instead of text. Can be combined only with color and size (the latter being line width, defaulting to 2)
+ * @param italic        Renders text in italic font style (not implemented)
+ * @param bold          Renders text in bold font style (not implemented)
+ * @param strike        Renders text in strikethrough font style (not implemented)
+ * @param shadow        Renders text with a dark shadow (not implemented)
+ * @param starred       Decorates text with a star icon - if set, it receives the [color] instead of the text.
+ * @param centered      Centers the line (and turns off wrap)
```

Comment:
I feel that much of this text is unnecessary. Where relevant, this should be on each field, and not on the class header.

Reply:
:+1: I wanted a complete documentation _somewhere_ and didn't think enough about the usefulness in terms of _how it looks in Studio_, and accessibility without actually opening CivilopediaText.kt. Tricky - changing `@param` to `@property` seems fine at first, then just `[mention]` the properties in the description shortly, and the class tooltip looks _much_ easier. But the individual properties - ugh:
   ![image](https://user-images.githubusercontent.com/63000004/124382663-cdd83280-dcc8-11eb-9692-2c903401d727.png)
I'd say that's a bug in the IDE - but how do we deal with that? Gimme some time.

Reply:
FYI: "`@property name﻿
Documents the property of a class which has the specified name. This tag can be used for documenting properties declared in the primary constructor, where putting a doc comment directly before the property definition would be awkward.`"
... But the "awkward" version is the one rendered properly in tooltips. So -  use `@property` and open another can of worms, create jetbrains account, start the war to get that ugliness fixed. ***or*** - use ugly source and get prettier studio tooltips now. I'll go for the latter.